2023-risks/m4-improper-credential-usage

```diff
@@ -0,0 +1,89 @@
+---
+
+layout: col-sidebar
+title: "M4: Improper Credential Usage"
+---
+
+# Threat Agents
+
+**Application Specific**
+
+Threat agents exploiting hardcoded credentials and improper credential usage in mobile applications can include automated attacks using publicly available or custom-built tools. Such agents could potentially locate and exploit hardcoded credentials or exploit weaknesses due to improper credential usage.
+
+# Attack Vectors
+
+**Exploitability EASY**
+
+Adversaries can exploit vulnerabilities in both hardcoded credentials and improper credential usage. Once these vulnerabilities are identified, an attacker can use hardcoded credentials to gain unauthorized access to sensitive functionalities of the mobile app. They can also misuse credentials, for instance by gaining access through improperly validated or stored credentials, thereby bypassing the need for legitimate access.
+
+# Security Weakness
+
+**Prevalence COMMON** <br />
+**Detectability EASY**
+
+Poor implementation of credential management, such as using hardcoded credentials and improper handling, can lead to severe security weaknesses. A comprehensive security testing process should aim to identify these issues. For instance, security testers should attempt to identify hardcoded credentials within the mobile app's source code or within any configuration files.
+
+Improper credential usage is another concern and may occur when mobile apps transmit credentials insecurely, store credentials insecurely on the device, or fail to adequately protect credentials during user authentication. Testers should evaluate credential transmission methods, storage practices, and the robustness of user authentication protocols to detect such vulnerabilities.
+
+
+# Technical Impacts
+
+**Impact SEVERE**
+
+Poor credential management can lead to a number of significant technical impacts. Unauthorized users might gain access to sensitive information or functionality within the mobile app or its backend systems. This can lead to data breaches, loss of user privacy, fraudulent activity, and potentially access to administrative functionality.
+
+# Business Impacts
+
+**Application / Business Specific**
+
+The business impact of poor credential management, including hardcoded credentials and improper credential usage, can be substantial:
+
+- Reputation Damage;
+- Information Theft;
+- Fraud;
+- Unauthorized Access to Data.
+
+# Am I Vulnerable To 'Insecure Authentication / Authorization'?
```

Comment:
Am I Vulnerable To 'Insecure Authentication / Authorization'? or Improper Credentials Usage?